test/compilable/test_ctfeonly.d

```diff
@@ -0,0 +1,20 @@
+// PERMUTE_ARGS:
+// REQUIRED_ARGS:
+// POST_SCRIPT: compilable/extra-files/ctfeonly-postscript.sh
+// I guess the platforms below don't have nm
+// DISABLED: win32 win64 osx
+
+
+string ctfeOnly(string x, string y)
+{
+    assert(__ctfe);
+    return (x ~ " " ~ y);
+}
+string ctfeOnlyIn(string x, string y)
+in {
+    assert(__ctfe);
+}
+do
```

Comment:
You might want to add another in-contract test case for the other possible syntax:

```D
string ctfeOnlyIn(string x, string y)
in (__ctfe)
{
}
```

... or is it not intended that this will work?